Cineaste/Models/StoredMovie.swift

```diff
@@ -20,9 +32,129 @@ class StoredMovie: NSManagedObject {
         if let moviePoster = movie.poster {
             poster = UIImageJPEGRepresentation(moviePoster, 1)
         }
-        voteAverage = movie.voteAverage
+        voteAverage = movie.voteAverage as NSDecimalNumber
+        voteCount = movie.voteCount
         runtime = movie.runtime
         releaseDate = movie.releaseDate
         watched = false
+        watchedDate = nil
+        listPosition = 0
+    }
+
+    convenience init(context moc: NSManagedObjectContext) {
+        let name = String(describing: type(of: self))
+        guard let entity = NSEntityDescription.entity(forEntityName: name, in: moc) else {
+            fatalError("Unable to create entity description with \(name)")
+        }
+
+        self.init(entity: entity, insertInto: moc)
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case overview
+        case listPosition
+        case releaseDate = "release_date"
+        case runtime
+        case voteAverage = "vote_average"
+        case voteCount = "vote_count"
+        case watched
+        case watchedDate
+        case id
+        case posterPath = "poster_path"
+        case title
+    }
+
+    required convenience init(from decoder: Decoder) throws {
```

Comment:
Really nice didn't know that you can combine `Codable` with `NSManagedObject`.
But we also have to take care because any ManagedObject is automatically part of CoreData and can either be stored persistently or cause Merge Conflict if used carelessly.